### PR TITLE
📆 Validate repeat_days_of_week when updating schedules

### PR DESCRIPTION
**Pull request by rowanseymour**
Trying to fix https://github.com/rapidpro/rapidpro/issues/1062 .. but I can't as yet reproduce the form submitting "NaN" ...

### CODE REVIEW

File: temba/schedules/views.py [hunk 1]
```diff
@@ -31,6 +32,15 @@ class ScheduleForm(BaseScheduleForm, forms.ModelForm):
     start = forms.CharField(max_length=16)
     start_datetime_value = forms.IntegerField(required=False)
 
+    def clean_repeat_days_of_week(self):
+        data = self.cleaned_data["repeat_days_of_week"]
+        if data:
+            for c in data:
+                if c not in Schedule.DAYS_OF_WEEK_OFFSET:
+                    raise forms.ValidationError(_("%(day)s is not a valid day of the week"), params={"day": c})
```
Comment (rowanseymour) [inline temba/schedules/views.py]: apparently this is the way to put vars in validation errors https://docs.djangoproject.com/en/2.2/ref/forms/validation/#raising-validationerror

File: temba/schedules/models.py [hunk 2]
```diff
@@ -163,18 +163,16 @@ def get_next_fire(cls, schedule, now):
 
             return next_fire
 
+        # if weekly, move forward until we're in the future and on an appropriate day of the week
         elif schedule.repeat_period == Schedule.REPEAT_WEEKLY:
             assert schedule.repeat_days_of_week != "" and schedule.repeat_days_of_week is not None
 
-            while (
-                next_fire <= now
-                or Schedule.DAYS_OF_WEEK_OFFSET[next_fire.weekday()] not in schedule.repeat_days_of_week
```
Comment (rowanseymour) [inline temba/schedules/models.py]: if `schedule.repeat_days_of_week` isn't a valid set of days, while loop runs until date overflows

**Comment by norkans7** [comment]
could the Nan be from flow imports? I think for campaigns import, I do not see that we import scheduled triggers

**Comment by rowanseymour** [comment]
I'm pretty sure `NaN` is a Javascript thing rather than a Python thing... so maybe at some point it was possible for the frontend code to generate that now is fixed

**Comment by rowanseymour** [comment]
Note that this was also the source of https://sentry.io/organizations/nyaruka/issues/1292880979/?project=1281372&query=is%3Aunresolved but I've checked and this schedule was fixed by the user and there aren't others like it

**Pull request merged**